api/src/main/java/org/openmrs/module/ModuleUtil.java

```diff
@@ -418,31 +424,24 @@ public static int compareVersion(String version, String value) {
 				return 0;
 			}
 			
-			List<String> versions = new ArrayList<>();
-			List<String> values = new ArrayList<>();
-			String separator = "-";
-			
-			// strip off any qualifier e.g. "-SNAPSHOT"
-			int qualifierIndex = version.indexOf(separator);
-			if (qualifierIndex != -1) {
-				version = version.substring(0, qualifierIndex);
-			}
-			
-			qualifierIndex = value.indexOf(separator);
-			if (qualifierIndex != -1) {
-				value = value.substring(0, qualifierIndex);
-			}
-			
-			Collections.addAll(versions, version.split("\\."));
-			Collections.addAll(values, value.split("\\."));
-			
-			// match the sizes of the lists
-			while (versions.size() < values.size()) {
-				versions.add("0");
-			}
-			while (values.size() < versions.size()) {
-				values.add("0");
+			Matcher versionMatcher = VERSION_PATTERN.matcher(version);
+			Matcher valueMatcher = VERSION_PATTERN.matcher(value);
+
+			if (!versionMatcher.matches() || !valueMatcher.matches()) {
+				// this seems wrong
+				return 0;
 			}
+
+			final String[] versions = {
+				versionMatcher.group("major"),
+				versionMatcher.group("minor"),
+				versionMatcher.group("patch")
+			};
+			final String[] values = {
+				valueMatcher.group("major"),
+				valueMatcher.group("minor"),
+				valueMatcher.group("patch")
+			};
 			
 			for (int x = 0; x < versions.size(); x++) {
```

Comment:
So now that we've changed `versions` and `values` from lists to arrays, we need adapt this look to use `versions.length` and, e.g., `versions[x]` to get teh value.